plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/PlayAssetDelivery.java

```diff
@@ -33,8 +40,23 @@
  */
 public class PlayAssetDelivery extends GodotPlugin {
 
+  private AssetPackManager assetPackManager;
+
+  /**
+   * Helper function that returns an AssetPackManager instance. Initially part of the constructor,
+   * refactored to provide easy access for writing mock tests.
+   *
+   * @param godot
+   * @return
+   */
+  AssetPackManager getAssetPackManagerInstance(Godot godot) {
```

Comment:
Rather than overriding this function in your tests, you could add another constructor:
```
AssetPackManager getAssetPackManagerInstance(Godot godot, AssetPackManager assetPackManager)
```

That way the test can provide the assetPackManager mock directly.

Reply:
Nice call! Having a separate constructor is a much better solution comparing to the previous one.